yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -291,9 +291,9 @@ export class AztecNodeService implements AztecNode {
    * @param leafValue - The value to search for
    * @returns The index of the given leaf in the given tree or undefined if not found.
    */
-  public async findLeafIndex(treeId: MerkleTreeId, leafValue: Buffer): Promise<bigint | undefined> {
+  public async findLeafIndex(treeId: MerkleTreeId, leafValue: Fr): Promise<bigint | undefined> {
     const committedDb = await this.#getWorldState();
-    return committedDb.findLeafIndex(treeId, leafValue);
+    return committedDb.findLeafIndex(treeId, leafValue.toBuffer());
```

Comment:
should the find leaf index api also use Fr ?

Reply:
I felt like changing the types in the merkle tree package could get a bit tricky in some places we pass the serialized indexed tree value which should not be Fr. Also we'll nuke the whole package and replace it with Rust or C++ so it's probably ok to keep the code disgusting.